app/validators/form_password_validator.rb

```diff
@@ -2,8 +2,20 @@ module FormPasswordValidator
   extend ActiveSupport::Concern
 
   included do
+    attr_accessor :password
+    attr_reader :user
+
     validates :password,
               presence: true,
               length: Devise.password_length
+
+    validate :valid_password
+  end
+
+  def valid_password
+    return if errors[:password].any?
+    temp_user = User.new(password: password)
+    return if temp_user.valid?
+    errors.add :password, temp_user.errors[:password]
```

Comment:
Agreed. I thought about this after writing my comments and came to the same conclusion.